diagnostics/stratonovich_diagonal.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import argparse
+import matplotlib.pyplot as plt
+import numpy as np
+import torch
+import tqdm
+from scipy import stats
+
+from tests.problems import Ex2
+from torchsde import sdeint, BrownianInterval
+from torchsde.settings import LEVY_AREA_APPROXIMATIONS
+from .utils import to_numpy, makedirs_if_not_found, compute_mse
+
+
+def inspect_sample():
+    batch_size, d = 32, 1
+    steps = 100
+
+    ts = torch.linspace(0., 5., steps=steps).to(device)
+    dt = 1e-1
+    y0 = torch.ones(batch_size, d).to(device)
+    sde = Ex2(d=d).to(device)
+    sde_strat = Ex2(d=d, sde_type='stratonovich').to(device)
+    sde_strat.p = sde.p
+    sde_strat.f = sde.f_corr
+
+    with torch.no_grad():
+        bm = BrownianInterval(t0=ts[0], t1=ts[-1], shape=y0.shape, dtype=y0.dtype, device=device,
+                              levy_area_approximation=LEVY_AREA_APPROXIMATIONS.space_time)
+
+        ys_euler = sdeint(sde, y0=y0, ts=ts, dt=dt, bm=bm, method='euler')
+        ys_heun = sdeint(sde_strat, y0=y0, ts=ts, dt=dt, bm=bm, method='heun')
```

Comment:
Can you add method=`midpoint` as well?

Reply:
Sure! Done
Rate plot: 
![](https://i.ibb.co/Jn0vhDC/rate.png)